typing/typetexp.ml

```diff
@@ -1066,6 +1066,10 @@ and transl_type_aux env ~row_context ~aliased ~policy mode styp =
       in
       let cty = transl_type new_env ~policy ~row_context mode t in
       ctyp (Ttyp_open (path, mod_ident, cty)) cty.ctyp_type
+  | Ptyp_of_kind jkind ->
+    let tjkind = jkind_of_annotation (Type_of_kind loc) styp.ptyp_attributes jkind in
+    let ty = Btype.newgenty (Tof_kind tjkind) in
```

Comment:
I wouldn't use `newgenty` here. Just use `newty`. Using `newgenty` will violate the invariant that levels always go down as you descend through a type.